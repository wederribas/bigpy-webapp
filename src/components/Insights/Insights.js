import React from 'react'

export default function Insights(props) {
  return <div>Insights</div>
}
