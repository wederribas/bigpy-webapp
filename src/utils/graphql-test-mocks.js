import { getCompaniesQuery } from '../components/Search/Search'

export const mocks = [
  {
    request: {
      query: getCompaniesQuery
    },
    result: {
      data: {
        allCompaniesNames: {
          edges: [
            { node: { Id: 'Company 0' } },
            { node: { Id: 'Company 1' } },
            { node: { Id: 'Company 2' } }
          ]
        }
      }
    }
  }
]
