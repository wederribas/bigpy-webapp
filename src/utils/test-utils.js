import React from 'react'
import { render } from 'react-testing-library'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
})

const customRender = node => {
  return render(<MuiThemeProvider theme={theme}>{node}</MuiThemeProvider>)
}

export * from 'react-testing-library'

export { customRender as render }
