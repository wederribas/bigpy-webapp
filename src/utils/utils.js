// @flow

/**
 * Deburr special Latin letters to basic Latin letters and removing
 * combining diacritical marks.
 */
export function deburr(string: string): string {
  return string.normalize('NFD').replace(/[\u0300-\u036f]/g, '')
}
