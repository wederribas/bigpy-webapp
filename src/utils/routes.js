import Search from '../components/Search/Search'
import Top10 from '../components/Top10/Top10'
import Insights from '../components/Insights/Insights'
import About from '../components/About/About'
import Dashboard from '../components/Dashboard/Dashboard'

export const routes = [
  {
    path: '/',
    component: Search,
    exact: true
  },
  {
    path: '/dashboard/:companyName',
    component: Dashboard
  },
  {
    path: '/top10',
    component: Top10
  },
  {
    path: '/insights',
    component: Insights
  },
  {
    path: '/about',
    component: About
  }
]
