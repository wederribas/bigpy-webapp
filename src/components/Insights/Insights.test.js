import React from 'react'
import { render } from '../../utils/test-utils'
import Insights from './Insights'

test('Insights page testing', () => {
  const { container, getByText } = render(<Insights />)

  expect(container.firstChild).toMatchSnapshot()
  expect(getByText('Insights')).toHaveTextContent('Insights')
})
