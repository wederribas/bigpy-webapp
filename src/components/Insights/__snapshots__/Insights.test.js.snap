// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insights page testing 1`] = `
<div>
  Insights
</div>
`;
